src/submission/submission-service.ts

```diff
@@ -1005,17 +1006,22 @@ export namespace operations {
     return undefined;
   };
 
-  export function mergeIcgcLegacyData(clinicalData: any, programId: string) {
+  export function mergeIcgcLegacyData(
+    legacySamples: Readonly<LegacyICGCImportRecord>[],
+    programId: string,
+  ) {
     const result: Donor[] = [];
-
-    clinicalData.donors.forEach((d: { [k: string]: string }, i: number) => {
-      validateRequiredColumns(i, d, ['icgc_donor_id', 'submitted_donor_id']);
+    legacySamples.forEach((sampleRow: LegacyICGCImportRecord, i: number) => {
+      validateRequiredColumns(i, sampleRow, ['icgc_donor_id', 'submitted_donor_id', 'donor_sex']);
       result.push({
```

Comment:
Isn't it possible to have duplicate donors in this result?? I see there are multiple rows in the file with same donorIds

Reply:
e.g. `DO35132`

Reply:
I'll check it

Reply:
It's not critical. It just the response JSON will fail on the import endpoint if there are duplicates.

Reply:
yes there will be duplicates, I didn't test the import manually, I submitted the  PR first :P